aead.go

```diff
@@ -131,6 +135,11 @@ func (g genericAead) Seal(dst, nonce, plaintext, additionalData []byte) []byte {
 			err = fmt.Errorf("C_Encrypt: %v", err)
 			return
 		}
+
+		if g.key.context.cfg.UseGCMIVFromHSM {
+			copy(nonce, params.IV())
```

Comment:
Should we return an error if `nonce` is smaller than `params.IV()` else the caller will never be able to access the generated value as it will have been truncated during `copy`.

Reply:
Good idea. Fixed in ad6f4b1.